backbone.js

```diff
@@ -1408,6 +1408,31 @@
     return Backbone.$.ajax.apply(Backbone.$, arguments);
   };
 
+  // A psuedo Promise implementation used to ensure asynchronous methods
+  // return thenables.
+  // Override this if you'd like to use a different ES6 library.
+  Backbone.Promise = function() {
```

Comment:
-1 for jquery fallback, the implementation differs to greatly from the
promise spec
On Jun 1, 2015 11:06 AM, "Justin Ridgewell" <notifications@github.com>
wrote:

> In backbone.js
> <https://github.com/jashkenas/backbone/pull/3651#discussion_r31434073>:
>
> > @@ -1408,6 +1408,31 @@
> >      return Backbone.$.ajax.apply(Backbone.$, arguments);
> >    };
> >
> > +  // A psuedo Promise implementation used to ensure asynchronous methods
> > +  // return thenables.
> > +  // Override this if you'd like to use a different ES6 library.
> > +  Backbone.Promise = function() {
>
> We could do a fallback system:
>
> if (root.Promise) {
>   BB.Promise = root.Promise;
> } else if (Backbone.$) {
>   BB.Promise = jQueryDeferredWrapper;
> } else {
>   BB.Promise = GiveHelpfulErrorMessageAKAYouAintGotNoPromises; // :wink:
> }
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/jashkenas/backbone/pull/3651/files#r31434073>.
>